File: internal/model/smartboard.go
package model

import "time"

// SmartBoardPanel represents a cached panel in the database
type SmartBoardPanel struct {
	ID            int       `json:"id"`
	PanelType     string    `json:"panel_type"`
	Data          string    `json:"data"` // JSON blob
	SessionID     string    `json:"session_id"` // OpenCode session ID for reuse
	LastRefreshed time.Time `json:"last_refreshed"`
	CreatedAt     time.Time `json:"created_at"`
}

// ThingsToRememberItem represents a single item in the Things to Remember panel
type ThingsToRememberItem struct {
	ID       string `json:"id"`
	Title    string `json:"title"`    // Short title (max 40 chars)
	Text     string `json:"text"`     // Full description/details
	Category string `json:"category"` // "urgent", "important", "not-important"
	Source   string `json:"source"`
	Date     string `json:"date"`
}

// ThingsToRememberData holds all items for the Things to Remember panel
type ThingsToRememberData struct {
	Items []ThingsToRememberItem `json:"items"`
}

// SuggestionItem represents a single coaching suggestion
type SuggestionItem struct {
	ID         string `json:"id"`
	Title      string `json:"title"`      // Short title (max 40 chars)
	Suggestion string `json:"suggestion"` // Full suggestion text
	Reasoning  string `json:"reasoning"`
	Status     string `json:"status"` // "active", "dismissed", "completed"
	CreatedAt  string `json:"createdAt"`
}

// SuggestionsData holds all suggestions for the Suggestions panel
type SuggestionsData struct {
	Suggestions []SuggestionItem `json:"suggestions"`
}

// AchievementItem represents a single achievement
type AchievementItem struct {
	ID          string `json:"id"`
	Title       string `json:"title"`       // Short title (max 40 chars)
	Achievement string `json:"achievement"` // Full achievement description
	Date        string `json:"date"`
	Source      string `json:"source"`
}

// AchievementsData holds all achievements for the Achievements panel
type AchievementsData struct {
	Achievements []AchievementItem `json:"achievements"`
}

// BlockerItem represents a single blocker
type BlockerItem struct {
	ID      string `json:"id"`
	Title   string `json:"title"`   // Short title (max 40 chars)
	Blocker string `json:"blocker"` // Full blocker description
	Context string `json:"context"`
	Date    string `json:"date"`
	Source  string `json:"source"`
}

// BlockersData holds all blockers for the Blockers panel
type BlockersData struct {
	Blockers []BlockerItem `json:"blockers"`
}

File: sidecar/schemas/smartboard.js
// JSON schemas for smartboard panel structured output
// Used with OpenCode SDK's format: { type: "json_schema", schema: {...} }

export const thingsToRememberSchema = {
  type: "array",
  items: {
    type: "object",
    properties: {
      id: { type: "string", description: "Item ID (empty string if new, reuse existing ID if same item)" },
      title: { type: "string", description: "Short concise title (max 40 chars)" },
      text: { type: "string", description: "Full description with context and details (max 200 chars)" },
      category: {
        type: "string",
        enum: ["urgent", "important", "not-important"],
        description: "Priority category: urgent (time-sensitive), important (high priority), not-important (nice to know)"
      },
      source: { type: "string", description: "Source filename where found" },
      date: { type: "string", description: "Date in YYYY-MM-DD format" }
    },
    required: ["id", "title", "text", "category", "source", "date"]
  }
}

export const suggestionsSchema = {
  type: "array",
  items: {
    type: "object",
    properties: {
      id: { type: "string", description: "Item ID (empty string if new, reuse existing ID if same item)" },
      title: { type: "string", description: "Short title (max 40 chars)" },
      suggestion: { type: "string", description: "Full actionable suggestion (max 150 chars)" },
      reasoning: { type: "string", description: "Why this matters and what pattern was observed (max 200 chars)" }
    },
    required: ["id", "title", "suggestion", "reasoning"]
  }
}

export const achievementsSchema = {
  type: "array",
  items: {
    type: "object",
    properties: {
      id: { type: "string", description: "Item ID (empty string if new, reuse existing ID if same item)" },
      title: { type: "string", description: "Short title (max 40 chars)" },
      achievement: { type: "string", description: "Full achievement description (max 200 chars)" },
      date: { type: "string", description: "Date in YYYY-MM-DD format" },
      source: { type: "string", description: "Source filename" }
    },
    required: ["id", "title", "achievement", "date", "source"]
  }
}

export const blockersSchema = {
  type: "array",
  items: {
    type: "object",
    properties: {
      id: { type: "string", description: "Item ID (empty string if new, reuse existing ID if same item)" },
      title: { type: "string", description: "Short title (max 40 chars)" },
      blocker: { type: "string", description: "Full blocker description (max 150 chars)" },
      context: { type: "string", description: "Additional details or who/what is involved (max 200 chars)" },
      date: { type: "string", description: "Date in YYYY-MM-DD format" },
      source: { type: "string", description: "Source filename" }
    },
    required: ["id", "title", "blocker", "context", "date", "source"]
  }
}

// Map panel type to schema
export const schemas = {
  "things-to-remember": thingsToRememberSchema,
  "suggestions": suggestionsSchema,
  "achievements": achievementsSchema,
  "blockers": blockersSchema
}
